### ISSUE TITLE
vet false positives on function print detection

### ISSUE THREAD

**Issue by Egojard** [golang/go#15099]
It seems vet will warn on any method named "Log" (and probably Printf etc.) whether they're the actual methods from the log package or not.

See http://play.golang.org/p/v6Hi0asMqy : it runs perfectly in playground which IIRC is still go1.5x, but generates an error in go1.6

1. What version of Go are you using (`go version`)?
go1.6 windows/amd64

2. What operating system and processor architecture are you using (`go env`)?
Windows 10, amd64

3. What did you do?
Passed a function value parameter to a method named "Log"
```
package main

func Log(callback func() int) int {
	return callback()
}

func main() {
	Log(func() int {
		return 42
	})
}
```

4. What did you expect to see?
Passed without error

5. What did you see instead?
```
[...]veterror\veterror.go:8: arg func() int {
           return 42
       } in Log call is a function value, not a function call
```

**Comment by robpike** [comment]
Working as intended, sorry. Vet has no good general way to tell that this argument to a function with a well-known name does not satisfy the usual conditions.

You can either ignore the warning, change the name of the function, or use the -printfuncs flag to vet to work around this.

**Comment by valyala** [comment]
@Egojard , FYI, this may be fixed with #15067 by [the CL](https://go-review.googlesource.com/#/c/21513/1)

**Comment by Egojard** [comment]
Cheers! No big issue at all, I indeed just renamed my methods, but it got me scratching my head a few minutes when migrating our code base.
